### DISCUSSION TITLE
Skip task if folder in git repo not exists

### DISCUSSION THREAD

**Discussion by KIRY4** [tektoncd/pipeline#5076] [Q&A] [answered]
### Feature request

Hello everyone. Is it possible somehow to skip step in pipeline if some folder not exists in git repo which was fetched in pervious step?

### Use case

We have several types of repos. With following folder structure in repo.

1) *-api, *-service, 
2) *-api, *-entity, *-service

For 1st type in my pipeline I have two gradle tasks with specific parameters for each task.
For 2nd type I want to add one more gradle task which will be run only if *-entity folder exists.

Maybe it is already to possible to do something like that with conditions or when?

**Comment by dibyom** [comment, marked as answer]
Yeah, your task can [emit a result](https://tekton.dev/docs/pipelines/tasks/#emitting-results) (e.g. this result be an array that contains a list of folder names that were fetched) that can then be checked in a when expression in the next task to decide whether it should run or not.
